docs/operator-manual/applicationset/Generators-Plugin.md

```diff
@@ -0,0 +1,237 @@
+# Plugin Generator
+
+Plugins allow you to provide your own generator.
+
+* You can write in any language
+* Simple: a plugin just responds to RPC HTTP requests.
+* You can use it in a sidecar, or standalone deployment.
+* You can get your plugin running today, no need to wait 3-5 months for review, approval, merge and an Argo software
+  release.
+* You can combine it with Matrix or Merge.
+
+## Simple example
+
+Using a generator plugin without combining it with Matrix or Merge.
+
+```yaml
+apiVersion: argoproj.io/v1alpha1
+kind: ApplicationSet
+metadata:
+  name: myplugin
+spec:
+  generators:
+    - plugin:
+        # The name of the plugin used
+        name: my-plugin
+        # Specify the configMap where the plugin configuration is located
+        configMapRef: my-plugin
+        # You can pass parameters included in the RPC call
+        params:
+          key1: "value1"
+          key1: "value1"
+        # When using a Plugin generator, the ApplicationSet controller polls every `requeueAfterSeconds` interval (defaulting to every 30 minutes) to detect changes.
+        requeueAfterSeconds: 30
+        # ...
+```
+
+* `name`: Required name of the Plugin.
+* `configMapRef`: A `ConfigMap` name containing the Plugin configuration to use for RPC call.
+* `params`: Parameters included in the RPC call. (Optional)
+
+!!! note
+    The concept of the plugin should not undermine the spirit of GitOps by externalizing data outside of Git. The goal is to be complementary in specific contexts.
+    For example, when using one of the PullRequest generators, it's impossible to retrieve parameters related to the CI (only the commit hash is available), which limits the possibilities. By using a plugin, it's possible to retrieve the necessary parameters from a separate data source and use them to extend the functionality of the generator.
+
+### Add a ConfigMap to configure the access of the plugin
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: my-plugin
+data:
+  token: $plugin.myplugin.token # Alternatively $<some_K8S_secret>:plugin.myplugin.token
+  baseUrl: http://myplugin.plugin.svc.cluster.local.
```

Comment:
For the purpose of the docs, should we just assume that the plugin is in the same ns as the controller and use `http://myplugin`?

Reply:
In my opinion, the plugin should be placed in a separate namespace, which allows for better control of the flows (netpol) and distinguishes their usage. In reality, it is the administrator's choice, depending on the organization's hierarchy. It's possible that separate DevOps teams, for example, maintain a plugin to manage CI/CD with separate parameters.

Reply:
I'm thinking about creating a template repo. For the purposes of that repo, do you think we should use a separate namespace, or would same ns be fine?

Reply:
Let's keep it simple for the documentation/template and use the ArgoCD namespace, which is fine with me.

We can still add a note to encourage segmentation.

Reply:
I went ahead with namespaced, since it is the preferred config, and people rarely stray from the boilerplate: https://github.com/argoproj-labs/applicationset-hello-plugin